src/index.js

```diff
@@ -36,7 +36,9 @@ ReactDOM.render(
       <Router history={history}>
         <Route path="/auth"  component={Auth} />
         <Route path="/" component={App} />
+        {/* Leave this for legacy support */}
         <Route path="/(:username)/snippets/(:id)" component={App} />
```

Comment:
Good idea. Couldn't get this exact implementation to work, but I added code to push the correct URL once the snippet has loaded.